index.d.ts

```diff
@@ -1,3 +1,9 @@
+type JSONObject = {[key: string]: JSONValue};
+interface JSONArray extends Array<JSONValue> {}
+export type JSONValue = string | number | boolean | null | JSONObject | JSONArray;
+
+export type JSONStringifyable = object | number | string | boolean;
```

Comment:
This seems wrong. `null` can be stringified, for example, and `ky.post(url, { json : null })` should work today.

Shouldn't the `json` option use `JSONValue` instead? And if I'm right, then this can be removed.

Reply:
Ah, missed the `null`.

The reason I didn't go with `JSONValue` is that it might be inconvenient. What if you have an object with a property with a value of `new Date()` and you just want it to be stringified? `JSON.stringify()` usage is expected to coerce values and I'm not sure we should be super strict here. Thoughts?

Reply:
So the idea is essentially to be explicit and "strict" about _output_ of `.json()`, but flexible about input of the `json` option. That makes sense to me. 👍